vignettes/decorate-modules-output.Rmd

```diff
@@ -0,0 +1,389 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` is a powerful `shiny`-based framework with built-in modules for interactive data analysis. 
+This document outlines the customization options available for modifying the output of `teal` modules. 
+You will learn how to use `teal_transform_module` to modify and enhance the objects created by `teal` modules, 
+enabling you to tailor the outputs to your specific requirements without rewriting the original module code.
+
+Adjusting input data or customizing module outputs in `teal` is accomplished using `transforms` created through 
+`teal_transform_module`. This document will focus on modifying outputs, and for clarity, we will refer to these 
+output-modifying objects as decoration objects or decorators.
+
+Throughout the examples, you will see references to a `data()` object. 
```

Comment:
`data()` has been extensively described in other vignette Including Data in teal Applications. I don't know if it is worth it to explain it again here

Reply:
Cool, will remove

Reply:
f9bdb60c692ffc465975f747232eee164b36a2d9